src/handlers.rs

```diff
@@ -68,3 +72,111 @@ pub async fn decode(req: web::Json<Request>) -> Result<impl Responder> {
     };
     Ok(web::Json(response))
 }
+
+async fn get_tx_logs(txn_hash: &DisplayBytes, network: &String) -> Result<Vec<TxLog>> {
+    let res = reqwest::get(format!(
+        "https://blockscout.com/{}/api?module=transaction&action=gettxinfo&txhash={}",
+        network, txn_hash
+    ))
+    .await
+    .map_err(|err| anyhow!(err))?;
+
+    if res.status() == StatusCode::NOT_FOUND {
+        return Err(Error::GetTxInfo("Wrong network".to_string()));
+    }
+
+    let res: Transaction = res.json().await.map_err(|err| anyhow!(err))?;
+
+    if res.status != "1" {
+        return Err(Error::GetTxInfo(res.message));
+    }
+
+    res.result
+        .map(|result| result.logs)
+        .ok_or_else(|| Error::Other(anyhow!("Missing input block")))
+}
+
+async fn get_contract_abi(network: &String, address: &DisplayBytes) -> Result<Contract> {
+    let res = reqwest::get(format!(
+        "https://blockscout.com/{}/api?module=contract&action=getabi&address={}",
+        network, address
+    ))
+    .await
+    .map_err(|err| anyhow!(err))?;
+
+    if res.status() == StatusCode::NOT_FOUND {
+        return Err(Error::GetTxInfo("Wrong network".to_string()));
+    }
+
+    let res: AbiResponse = res.json().await.map_err(|err| anyhow!(err))?;
+
+    if res.status == "0" {
+        return Err(Error::GetTxInfo(res.message));
+    }
+
+    let c: Contract = serde_json::from_str(
+        &res.result
+            .ok_or_else(|| anyhow!("Abi parsing failed".to_string()))?,
```

Comment:
I don't think you need `.to_string()` here, and also you probably can change it to just `ok_or(anyhow...)`